packages/flutter/lib/src/widgets/image.dart

```diff
@@ -822,6 +835,38 @@ class Image extends StatefulWidget {
   /// {@animation 400 400 https://flutter.github.io/assets-for-api-docs/assets/widgets/loading_progress_image.mp4}
   final ImageLoadingBuilder loadingBuilder;
 
+  /// A builder function that is called if an error occurs during image loading.
+  ///
+  /// If this builder is not provided, any exceptions will be reporetd to
+  /// [FlutterError.onError]. If it is provided, the caller should either handle
+  /// or rethrow the exception.
+  ///
+  /// {@tool dartpad --template=stateless_widget_material}
+  ///
+  /// The following sample uses [errorBuilder] to show a '😢' in place of the
+  /// image that fails to load, and prints the error to the console.
+  ///
+  /// ```dart
+  /// Widget build(BuildContext context) {
+  ///   return DecoratedBox(
+  ///     decoration: BoxDecoration(
+  ///       color: Colors.white,
+  ///       border: Border.all(),
+  ///       borderRadius: BorderRadius.circular(20),
+  ///     ),
+  ///     child: Image.network(
+  ///       'https://example.does.not.exist/image.jpg',
+  ///       errorBuilder: (BuildContext context, Object exception, StackTrace stackTrace) {
+  ///         print('An error occurred loading "https://example.does.not.exist/image.jpg": $exception');
```

Comment:
I would just remove the print. In general it wouldn't be good practice to have one here in an app and examples should show good practice.

Reply:
Replaced the print with a comment indicating how you might use this for analytics purposes.